esx_service/vmdk_ops.py

```diff
@@ -1636,7 +1636,12 @@ def main():
 
         kv.init()
         connectLocalSi()
+
+        # start the daemon. Do all the task to start the listener through the daemon
+        threadutils.start_new_thread(target=vm_listener.start_vm_changelistener,
+                                 daemon=True)
```

Comment:
connectLocalSi succeeded, we would get si and loop won't be executed. If not, then we need to retry to get si and only then start the listener. When host is restarted, hostd takes time to comeup and we don't get a connection to hostd instantly.